### PR TITLE
applications: nrf_desktop: Scan management

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds scan management (fix mouse pointer lag problems).

### CODE REVIEW

File: applications/nrf_desktop/src/modules/Kconfig.ble [hunk 1]
```diff
@@ -123,6 +130,22 @@ config DESKTOP_BLE_SCANNING_ENABLE
 	help
 	  Enable device to scan for peripheral devices.
 
+config DESKTOP_BLE_SCAN_START_TIMEOUT
```
Comment (SebastianBoe) [inline applications/nrf_desktop/src/modules/Kconfig.ble]: What do you think about

DESKTOP_BLE_SCAN_START_TIMEOUT_S

and

DESKTOP_BLE_SCAN_DURATION_S

?

Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/Kconfig.ble]: Ok, I may change it.

File: applications/nrf_desktop/src/modules/ble_scan.c [hunk 2]
```diff
@@ -132,29 +156,77 @@ static void scan_start(void)
...
-	LOG_INF("Scan started");
-	scanning = true;
+
+	k_delayed_work_submit(&scan_stop_trigger, SCAN_DURATION_MS);
+	k_delayed_work_cancel(&scan_start_trigger);
+
 	return;
 
 error:
 	LOG_ERR("Scanning failed to start (err %d)", err);
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/ble_scan.c]: this log seems excessive

File: applications/nrf_desktop/src/modules/ble_scan.c [hunk 3]
```diff
@@ -132,29 +156,77 @@ static void scan_start(void)
...
+	} else {
+		k_delayed_work_submit(&scan_start_trigger, SCAN_TRIG_CHECK_MS);
+	}
+}
+
+static void scan_stop_trigger_fn(struct k_work *w)
+{
+	static_assert(SCAN_DURATION_MS > 0, "");
+
+	struct bond_find_data bond_find_data = {
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/ble_scan.c]: it seems we use this in multiple places now
maybe we should create header in utils and create a incline function there:
`size_t get_conn_count() {}`

File: applications/nrf_desktop/src/modules/ble_scan.c [hunk 4]
```diff
@@ -202,12 +277,16 @@ static void discovery_completed(struct bt_gatt_dm *dm, void *context)
 
 static void discovery_service_not_found(struct bt_conn *conn, void *context)
 {
-	LOG_WRN("Cannot find service during the discovery");
+	LOG_ERR("Cannot find service during the discovery");
+	__ASSERT_NO_MSG(discovering_peer_conn == conn);
+	bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/ble_scan.c]: no unref?

Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_scan.c]: discovering_peer_conn  = NULL

Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_scan.c]: if this is handled from EM event please add a comment here

Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_scan.c]: It is handled from Event Manager (PEER_DISCONNECTED event). Added comment.

File: applications/nrf_desktop/src/modules/ble_scan.c (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_scan.c]: ok, removed

File: applications/nrf_desktop/src/modules/ble_scan.c (hunk 3 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_scan.c]: I will change it after we add reading VID and PID.
Let's leave it as it is now.

Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_scan.c]: discussed offline

**Pull request merged**